daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -304,7 +305,14 @@ class DataProcessor(val ssrd: SchemaSetRuntimeData)
   }
 
   def unparse(inputter: InfosetInputter, outStream: java.io.OutputStream) = {
-    val out = DirectOrBufferedDataOutputStream(outStream, null) // null means no other stream created this one.
+    val out = DirectOrBufferedDataOutputStream(
+      outStream,
+      null, // null means no other stream created this one.
+      isLayer=false,
+      tunablesObj.outputStreamChunkSizeInBytes,
+      tunablesObj.maxByteArrayOutputStreamBufferSizeInBytes,
+      Paths.get(tunablesObj.tempFilePath))
```

Comment:
Might be advantageous to modify the TunableGenerator so that the tempFilePath variable is of type File so we don't have to duplicate this conversion here and below (and anywhere else it might eventually be used). Also makes it easier to catch any conversion errors in the setTunable logic rather than where it's actuall used.